python/tests/t_request_header.py

```diff
@@ -0,0 +1,58 @@
+from kat.harness import variants, Query
+from abstract_tests import AmbassadorTest, ServiceType, HTTP
+
+class XRequestIdHeaderPreserveTest(AmbassadorTest):
+    target: ServiceType
+
+    def init(self):
+        self.target = HTTP(name="target")
+
+    def config(self):
+        yield self.target, self.format("""
+---
+apiVersion: ambassador/v0
+kind:  Module
+name:  ambassador
+config:
+  preserve_external_request_id: true
+---
+apiVersion: ambassador/v2
+kind:  Mapping
+name:  {self.name}-target
+prefix: /target/
+service: http://{self.target.path.fqdn}
+""")
+
+    def queries(self):
+        yield Query(self.url("target/"), headers={"x-request-id": "hello"})
+
+    def check(self):
+        assert self.results[0].backend.request.headers['x-request-id'] == ['hello']
+
+class XRequestIdHeaderDefaultTest(AmbassadorTest):
+    target: ServiceType
+
+    def init(self):
+        self.target = HTTP(name="target")
+
+    def config(self):
+        yield self.target, self.format("""
+---
+apiVersion: ambassador/v0
```

Comment:
I think that you should delete this `ambassador` module entirely from this test.

You should be able to run these tests locally by creating a cluster, setting `DEV_KUBECONFIG` and `DEV_REGISTRY` as described in `DEVELOPING.md`, then

```
make clean push pytest-only PYTEST_ARGS="-k XRequestId"
```

Does that pass for you?

Reply:
😨 

I didn't run the whole test suite locally, to be honest, and I couldn't parse if the build failures in Travis were 'real' or not. I'll take a look at this and get back to you.

Reply:
@kflynn test is breaking in my local too. Also if I revert the change and run this test it still breaks. Seems like this is not the correct place to right this test, thoughts?

Reply:
as discussed @kflynn I have set the default test as `xfail`. It passes rest of the tests.